bundles/action/org.openhab.action.squeezebox/src/main/java/org/openhab/action/squeezebox/internal/Squeezebox.java

```diff
@@ -144,6 +144,156 @@ public static boolean squeezeboxPlayUrl(
+			@ParamDoc(name = "url", text = "The URL to play (if empty will clear the playlist)") String url,
+			@ParamDoc(name = "volume", text = "The volume to set the device when playing this URL (between 1-100)") int volume,
+			@ParamDoc(name = "resumePlayback", text = "Continue playback after speech") Boolean resumePlayback ){
+		if (!isReady()) return false;
+		
+		// get the player - can return null if the playerId is bogus
+		SqueezePlayer player = squeezeServer.getPlayer(playerId);
+		if (player == null) 
+			return false;
+		// set the player ready to play this URL
+
+   		if (StringUtils.isEmpty(url)) {
+   			return squeezeServer.clearPlaylist(playerId);
+   		} else {
+			logger.trace("***START URL**** Player: '{}'", playerId);
+			// get the current player state
+			int playerVolume = player.getUnmuteVolume();
+			boolean playerPowered = player.isPowered();
+			boolean playerMuted = player.isMuted();
+			Mode playerMode = player.getMode();
+			int currNumTracks = player.getNumberPlaylistTracks();
+			int currPlayingTime = player.getCurrentPlayingTime();
+			int currPlaylistIndex = player.getCurrentPlaylistIndex();
+			int currPlaylistShuffle = player.getCurrentPlaylistShuffle();
+			int currPlaylistRepeat = player.getCurrentPlaylistRepeat();
+			int newNumTracks = 0;
+			logger.trace("Current Playing Mode '{}'", playerMode.toString());
+			logger.trace("Current Volume '{}'", playerVolume);
+			logger.trace("Current Num Playlist Tracks '{}'", currNumTracks);
+			logger.trace("Current Playing Playlist Index '{}'", currPlaylistIndex);
+			logger.trace("Current Playing Time '{}'", currPlayingTime);
+			logger.trace("Current Shuffle Mode '{}'", currPlaylistShuffle);
+			logger.trace("Current Repeat Mode '{}'", currPlaylistRepeat);
+			
+			// If Playing Pause player before adjusting volume!
+			if (playerMode == Mode.play)
+				squeezeServer.pause(playerId);
+			
+			// set the player ready to play this announcement
+			if (playerMuted) {
+				logger.trace("Setting player state: unmuted");
+				squeezeServer.unMute(playerId);
+			}
+			if (volume != -1) {
+				logger.trace("Setting player state: volume {}", volume);
+				squeezeServer.setVolume(playerId, volume);
+			}
+			
+			if (currPlaylistRepeat != 0)
+				squeezeServer.setRepeatMode(playerId, 0);
+			if (currPlaylistShuffle != 0) {
+				squeezeServer.setShuffleMode(playerId, 0);
+				currPlaylistIndex = 0;
+				logger.trace("Shuffle Changed! Set Current Playing Index to 0");
+			}
+			
+			SqueezeboxSentenceListener listener = new SqueezeboxSentenceListener(playerId);
+			squeezeServer.addPlayerEventListener(listener);
+			
+			// send the URL (this will power up the player and un-mute if necessary)
+			logger.trace("Adding URL to current playlist '{}' to play", url);
+			squeezeServer.addPlaylistItem(playerId, url);
+			logger.trace("Sleeping for 1s for updated playlist to refresh", url);
+			try {
+				Thread.sleep(1000);
+			} catch (InterruptedException e) {	}
```

Comment:
it could be worth adding an internal pauseThread() method in order to avoid the try-catch-blocks